src/app/mi-details/add-problem-statement/add-problem-statement.page.ts

```diff
@@ -0,0 +1,110 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { LoaderService } from 'src/app/services/loader/loader.service';
+import { ProjectsApiService } from 'src/app/services/projects-api/projects-api.service';
+import { ToastService } from 'src/app/services/toast/toast.service';
+import { finalize } from 'rxjs';
+import  urlConfig  from 'src/app/config/url.config.json'
+
+@Component({
+  selector: 'app-add-problem-statement',
+  templateUrl: './add-problem-statement.page.html',
+  styleUrls: ['./add-problem-statement.page.scss'],
+})
+export class AddProblemStatementPage implements OnInit {
+  headerConfig:any = {
+    showBackButton:true
+  }
+  showInput = false;
+  projectTemplateId:any;
+  showLoading:boolean = true;
+  problemStatement: string = '';
+  selectedOption: string = '';
+  options :any;
+  isRadioDisabled: boolean = false;
+  isActivatedProgram:any=true;
+  constructor(private router: Router,
+    private route:ActivatedRoute,
+    private toastService :ToastService,
+    private loader : LoaderService,
+    private ProjectsApiService: ProjectsApiService
+  ) { 
+      this.route.params.subscribe(param=>{
+        this.projectTemplateId = param['id'];
+      })
+  }
+
+  ngOnInit() {
+    this.addProblemStatementOptions()
```

Comment:
Give relevant function names based on its actions
Change it to getProblemStatementList